assets/js/editor-components/product-tag-control/index.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useState, useEffect, useCallback, useMemo } from '@wordpress/element';
+import { SearchListControl } from '@woocommerce/editor-components/search-list-control';
+import { SelectControl } from '@wordpress/components';
+import { getSetting } from '@woocommerce/settings';
+import { useDebouncedCallback } from 'use-debounce';
+
+/**
+ * Internal dependencies
+ */
+import type { SearchListItem as SearchListItemProps } from '../search-list-control/types';
+import ProductTagItem from './product-tag-item';
+import type { ProductTagControlProps } from './types';
+import { getProductTags } from '../utils';
+import './style.scss';
+
+/**
+ * Component to handle searching and selecting product tags.
+ */
+const ProductTagControl = ( {
+	isCompact = false,
+	onChange,
+	onOperatorChange,
+	operator = 'any',
+	selected,
+}: ProductTagControlProps ): JSX.Element => {
+	const [ list, setList ] = useState< SearchListItemProps[] >( [] );
+	const [ loading, setLoading ] = useState( true );
+	const [ isMounted, setIsMounted ] = useState( false );
+	const limitTags = getSetting( 'limitTags', false );
+
+	const selectedTags = useMemo< SearchListItemProps[] >( () => {
+		return list.filter( ( item ) => selected.includes( item.id ) );
+	}, [ list, selected ] );
+
+	const onSearch = useCallback(
+		( search: string ) => {
+			setLoading( true );
+			getProductTags( { selected, search } )
+				.then( ( newList ) => {
+					setList( newList );
+					setLoading( false );
+				} )
+				.catch( () => {
+					setLoading( false );
```

Comment:
We set the `list` state in the previous code to an empty array inside the `catch` block. Shouldn't we do the same here?

Reply:
I decided not to. It seemed like keeping a list visible on fail would be more graceful that showing an empty state with no items.